Lib/idlelib/idle_test/test_debugger.py

```diff
@@ -23,7 +34,456 @@ def test_init(self):
+
+        idb.clear_break.assert_called_once_with('test.py', 4)
+
+    def test_clear_file_breaks(self):
+        """Test the .clear_file_breaks() method calls idb."""
+        pyshell = make_pyshell_mock()
+        idb = mock.Mock()
+        test_debugger = debugger.Debugger(pyshell, idb)
+
+        test_debugger.clear_file_breaks('test.py')
+
+        idb.clear_all_file_breaks.assert_called_once_with('test.py')
+
+    def test_load_breakpoints(self):
+        """Test the .load_breakpoints() method calls idb."""
+        FileIO = namedtuple('FileIO', 'filename')
+
+        class MockEditWindow(object):
+            def __init__(self, fn, breakpoints):
+                self.io = FileIO(fn)
+                self.breakpoints = breakpoints
+
+        pyshell = make_pyshell_mock()
+        pyshell.flist = mock.Mock()
+        pyshell.flist.inversedict = (
+            MockEditWindow('test1.py', [1, 4, 4]),
+            MockEditWindow('test2.py', [13, 44, 45]),
+        )
+        idb = mock.Mock()
+        test_debugger = debugger.Debugger(pyshell, idb)
+
+        test_debugger.load_breakpoints()
+
+        idb.set_break.assert_has_calls(
+            [mock.call('test1.py', 1),
+             mock.call('test1.py', 4),
+             mock.call('test1.py', 4),
+             mock.call('test2.py', 13),
+             mock.call('test2.py', 44),
+             mock.call('test2.py', 45)])
+
+
+class StackViewerTest(unittest.TestCase):
+
+    def test_init(self):
+        """Test creation of StackViewer."""
+        gui = None
+        flist = None
+        master_window = None
+        sv = debugger.StackViewer(master_window, flist, gui)
+        self.assertTrue(hasattr(sv, 'stack'))
+
+    def test_load_stack(self):
+        """Test the .load_stack() method against a fixed test stack."""
+        # Arrange a test stack.
+        test_code = compile(TEST_CODE, 'test_load_stack.py', 'exec')
+        test_stack = [
+            (MockFrameType(test_code, 1), 1),
+            (MockFrameType(test_code, 2), 2)
+        ]
+
+        # Create a stackviewer and load the test stack.
+        test_sv = debugger.StackViewer(None, None, None)
+        test_sv.load_stack(test_stack)
+
+        # Check the test stack is assigned and the list contains the repr of them.
+        self.assertEquals(test_sv.stack, test_stack)
+        self.assertEquals(test_sv.get(0), '?.<module>(), line 1: ')
+        self.assertEquals(test_sv.get(1), '?.<module>(), line 2: ')
+
+    def test_show_source(self):
+        """Test the .show_source() method against a fixed test stack."""
+        # Arrange a test stack.
+        test_code = compile(TEST_CODE, 'test_show_source.py', 'exec')
+        test_stack = [
+            (MockFrameType(test_code, 1), 1),
+            (MockFrameType(test_code, 2), 2)
+        ]
+
+        # Create a stackviewer and load the test stack.
+        test_sv = debugger.StackViewer(None, None, None)
+        test_sv.load_stack(test_stack)
```

Comment:
Same note about `setUp()` and `tearDown()`.

Reply:
fixed